core/src/test/java/org/hibernate/ogm/utils/TestHelper.java

```diff
@@ -289,17 +295,37 @@ private static AvailableDatastoreProvider getDatastoreProvider() {
 			// This ignores the case where the provider is given as class or FQN; That's ok for now, can be extended if
 			// needed
 			Object datastoreProviderProperty = getDefaultTestStandardServiceRegistry( Collections.<String, Object>emptyMap() )
-				.getService( ConfigurationService.class )
-				.getSettings()
-				.get( OgmProperties.DATASTORE_PROVIDER );
-
-			AvailableDatastoreProvider provider = datastoreProviderProperty != null ? AvailableDatastoreProvider.byShortName( datastoreProviderProperty.toString() ) : null;
-
-			if ( provider == null ) {
+					.getService( ConfigurationService.class )
+					.getSettings()
+					.get( OgmProperties.DATASTORE_PROVIDER );
+
+			if ( datastoreProviderProperty != null ) {
+				String value = datastoreProviderProperty.toString();
+				if ( AvailableDatastoreProvider.isShortName( value ) ) {
+					return AvailableDatastoreProvider.byShortName( value );
+				}
+
+				for ( AvailableDatastoreProvider provider : AvailableDatastoreProvider.values() ) {
+					if ( provider.getDatastoreProviderClassName().equals( value ) ) {
+						return provider;
+					}
+				}
+
+				Class<?> configuredProviderClass = loadClass( value );
+				if ( configuredProviderClass != null ) {
+
+					for ( AvailableDatastoreProvider provider : AvailableDatastoreProvider.values() ) {
+						Class<?> availableProviderClass = loadClass( provider.getDatastoreProviderClassName() );
+
+						if ( availableProviderClass.isAssignableFrom( configuredProviderClass ) ) {
```

Comment:
How come does it need to be assignable to the available datastore provider? Can't we just instantiate it?

This function is also quite long, I wonder if a one line comment for each block wouldn't make it clearer what's going on.